### PR TITLE
Voice to Voice PR

### PR DESCRIPTION
**Pull request by square-1111**
**Moshi** - For initial setup and structuring the code
- [ ] Inference script 
- [ ] Evaluation script

### CODE REVIEW

File: constants/__init__.py [hunk 1]
```diff
@@ -29,9 +29,13 @@ class CompetitionParameters:
 # Schedule of model architectures
 COMPETITION_SCHEDULE: List[CompetitionParameters] = [
     CompetitionParameters(
-        reward_percentage=1.0,
+        reward_percentage=0.5,
         competition_id="o1",
     ),
+    CompetitionParameters(
+        reward_percentage=0.5,
+        competition_id="v1_moshi",
```
Comment (salmanshah1d) [inline constants/__init__.py:37]: please make `v1` only as discussed

File: miner_utils/upload_model.py [hunk 2]
```diff
@@ -84,12 +84,13 @@ def regenerate_hash(namespace, name, epoch, competition_id):
...
-def validate_repo(ckpt_dir, epoch):
-    for filename in get_required_files(epoch):
+def validate_repo(ckpt_dir, epoch, model_type):
+    for filename in get_required_files(epoch, model_type):
         filepath = os.path.join(ckpt_dir, filename)
         if not os.path.exists(filepath):
             raise FileNotFoundError(f"Required file {filepath} not found in {ckpt_dir}")
-    check_config(ckpt_dir)
+    if model_type == "o1":
+        check_config(ckpt_dir)
```
Comment (salmanshah1d) [inline miner_utils/upload_model.py:94]: should we include some checking logic similar to `check_config` function for `v1` competition?

File: model/storage/hugging_face/hugging_face_model_store.py [hunk 3]
```diff
@@ -79,7 +91,7 @@ async def upload_model(
         api = HfApi(token=token)
         export_readme(model.local_repo_dir)
         export_hotkey(model.local_repo_dir, hotkey)
-        files_to_upload = get_required_files(model.id.epoch) + [README_FILE] + [HOTKEY_FILE]
+        files_to_upload = get_required_files(model.id.epoch, model.id.competition_id.split("_")[-1]) + [README_FILE] + [HOTKEY_FILE]
```
Comment (salmanshah1d) [inline model/storage/hugging_face/hugging_face_model_store.py:94]: needs to change as discussed

File: neurons/validator.py [hunk 4]
```diff
@@ -1259,7 +1285,7 @@ async def run(self):
                     # Sleep for 5 minutes before resycing metagraph
                     await asyncio.sleep(60 * 5)
                 else:
-                    await self.try_run_step(ttl=60 * 15) # 15 minute timeout. Same as the timeout for resetting stale models being scored.
+                    await self.try_run_step(ttl=60 * 30) # 30 minute timeout. Same as the timeout for resetting stale models being scored.
```
Comment (salmanshah1d) [inline neurons/validator.py:1288]: need to put this back to 15

File: req_aud.txt
Comment (salmanshah1d) [inline req_aud.txt:1]: please remove this additional `req_aud.txt` file and make sure that all requirements are covered within the single `requirements.txt` file

File: neurons/model_scoring.py [hunk 5]
```diff
@@ -19,7 +19,7 @@
 
 HF_DATASET = "omegalabsinc/omega-multimodal"
 DATA_FILES_PREFIX = "default/train/"
-MIN_AGE = 4 * 60 * 60  # 4 hours
+MIN_AGE = 1 * 20 * 60  # 4 hours # changed from 4 hours to 20 minutes for testing
```
Comment (salmanshah1d) [inline neurons/model_scoring.py:22]: Need to put this back to 4 hours

File: neurons/validator.py [hunk 6]
```diff
@@ -950,14 +953,37 @@ async def run_step(self):
...
+                                time.sleep(MINS_TO_SLEEP * 60)
+                            score = get_model_score(
+                                hf_repo_id,
+                                mini_batch=eval_data,
+                                local_dir=self.temp_dir_cache.get_temp_dir(hf_repo_id),
+                                hotkey=hotkey,
+                                block=model_i_metadata.block,
+                                model_tracker=self.model_tracker
+                            )
+                        elif competition_parameters.competition_id == "v1_moshi":
```
Comment (salmanshah1d) [inline neurons/validator.py:971]: should be same `v1` here

File: neurons/validator.py
Comment (salmanshah1d) [inline neurons/validator.py:1]: In `is_model_old_enough` function, need to use `MIN_AGE` when competition = `o1` but need to use `V2V_MIN_AGE` when competition = `v1`

File: miner_utils/upload_model.py (hunk 2 above)
Comment (square-1111) [reply miner_utils/upload_model.py:94]: Yes, we need to include a training config once the training script is ready, as well as an inference config for the model name, version, and other details.

**Pull request merged**